src/components/Footer.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+import { Todo } from '../types/Todo';
+import { Status } from '../types/Status';
+import { Filter } from './Filter';
+import { Errors } from '../types/Errors';
+import { wait } from '../utils/fetchClient';
+
+type Props = {
+  onFilter: (value: Status) => void;
+  currentFilterStatus: Status;
+  todos: Todo[];
+  onDeleteTodo: (todoId: number) => Promise<unknown>;
+  clearError: () => void;
+  onSetError: (error: Errors) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  onFilter,
+  currentFilterStatus,
+  todos,
+  onDeleteTodo,
+  clearError,
+  onSetError,
+}) => {
+  const uncompletedTodos = todos.filter(item => !item.completed);
+
+  const isAnyTodoCompleted = todos.some(todo => todo.completed);
+
+  const handleClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    try {
+      await Promise.all(completedTodos.map(todo => onDeleteTodo(todo.id)));
+    } catch {
+      onSetError(Errors.Delete);
+    } finally {
+      wait(3000).then(() => clearError());
+    }
```

Comment:
In the positive case you don't have an error, so you don't need to remove it. And as I mentioned previously you do the same so many times so create a function